### ISSUE TITLE
an error with the training set direction

### ISSUE THREAD

**Issue by i-ting4931** [MCG-NJU/MeMOTR#22]
Hello, I recently installed this model to train a custom dataset. The environment setup is complete, and I first attempted to use the MOT17 dataset to test whether the training process works properly. However, during the training, I encountered some abnormal data, and I was wondering if you could provide any guidance on how to resolve this issue.

Currently, I have downloaded both the Crowdhuman and MOT17 datasets. However, while training, I noticed that all the loss values are zero, which seems to suggest that the data is not being properly loaded. To check the data loading path, I added the following line of code: print(f"Frame path: {frame_path}"). The result shows that the dataset is loading Crowdhuman, but when I issued the command, I set the dataset to MOT17. I'm not entirely sure where the problem lies—could you kindly take a look? Thank you very much.

Also, just to mention, my computer has only one GPU: an NVIDIA GeForce RTX 3060. Since my GPU is limited, do I need to modify any lambda functions in the code? I appreciate your help.

If there's anything that I didn't explain clearly, please feel free to let me know, and I will provide any additional details you may need.
Thank you again.

![image](https://github.com/user-attachments/assets/400363ea-2880-47a9-87a5-87558c2d739c)
![image](https://github.com/user-attachments/assets/49b6bd18-3e6b-4d49-a8ac-081287d15e8b)
![image](https://github.com/user-attachments/assets/5215e986-cac7-411b-ad30-8c8e88cb715a)

---------------------------------------------------------------------------------
Here is the related information regarding the failed training.
![image](https://github.com/user-attachments/assets/a0471f21-13b8-48a4-8cc3-24fb22530ec7)
![image](https://github.com/user-attachments/assets/c617fd6b-d7e4-4984-b9ff-147d67adaffc)

**Comment by i-ting4931** [comment]
Here is the place where I added the code I mentioned earlier.
![image](https://github.com/user-attachments/assets/2fdb0a03-94bd-4637-ae8c-26a008410b71)

**Comment by HELLORPG** [comment]
The data loading process is not related to the GPU. Therefore, I think you should not modified any data loading function on your GPU (3060).

I have never seen this issue before. Have you run the data preparation scripts (like `./data/gen_crowdhuman_gts.py`) before running the training script?

**Comment by i-ting4931** [comment]
Hello, considering that there might have been some mistakes in my previous operations, I deleted the previously cloned model and cloned it again. This time, I only modified the GPU quantity, downloaded the pre-trained weights and datasets, and placed them in the designated locations. I also used the gen_mot17_gts.py and gen_crowdhuman_gts.py files to generate the necessary files.

The training process is now running, but the loss values seem abnormal (most loss values start at 0 during the initial training). I suspect that the images might not have been successfully read by the model for training. Could you kindly advise if I made any mistakes? If the training were running correctly, what would the expected behavior look like? Thank you very much.
![image](https://github.com/user-attachments/assets/4644744d-bacc-469b-ae7b-3144d3d703d8)

**Comment by HELLORPG** [comment]
To be honest, this is really weird. I need to wait until I have a spare GPU server to show what the correct logging looks like. 

You could check the content of the data loaded into the training loop, after these lines:
https://github.com/MCG-NJU/MeMOTR/blob/7de13f4afee57f8d5fcf47567d890af401815174/train_engine.py#L192-L199
You can add code like these:
```python
print(batch["infos"][0])                          # the GTs
print(batch["imgs"][0][0].shape)                  # the image's shape
# Or others
```
You could analyze it based on the results or upload it here.

**Comment by i-ting4931** [comment]
Hello, I followed your suggestion and added the print statement. The current output is shown in the image below (Image 1). The prediction result tensor([], dtype=torch.int64) shows 0, and ids, areas, and labels all indicate that no objects were detected. I'm not quite sure why this is happening.

I noticed that in the train_mot17.yaml file, there is a setting "USE_CROWDHUMAN: True". I initially suspected that the issue might be due to training with MOT17 while having Crowdhuman included in the configuration. So, I changed "USE_CROWDHUMAN: True" to false, but this resulted in an error (Image 2).

I also tried some commands related to Submit and Evaluation, but I encountered a small issue. When using eval mode, I got the following error (Image 3), even though I don't have any related files. Could you please advise if this file is supposed to be generated automatically? If so, did I make a mistake somewhere?

Sorry for the multiple questions, and I truly appreciate your help.

Thank you very much.

(Image 1)
![image](https://github.com/user-attachments/assets/d257bfb7-dd1a-481b-8acf-a0a6b52723df)

(Image 2)
![image](https://github.com/user-attachments/assets/72c4ae36-cda7-4d6f-b83d-506b4131e278)

(Image 3)
![image](https://github.com/user-attachments/assets/3492522a-9b94-49b9-a59c-d6f0778c0061)

**Comment by HELLORPG** [comment]
According to (Image 2), it seems that you did not successfully load any image and annotation from MOT17. You can add some breakpoints during the data loading process to determine where the problem is.

For example, here:
https://github.com/MCG-NJU/MeMOTR/blob/7de13f4afee57f8d5fcf47567d890af401815174/data/mot17.py#L59-L68 all GTs should be loaded.

**Comment by i-ting4931** [comment]
Here’s a polite translation:

Hello,

I am currently encountering an issue during training where the images are found, but the corresponding bounding boxes are not correctly located. I have verified that the image and text file paths in mot17.py are correct. I have also tried various methods to resolve this issue, and I found that when generating text files using gen_crowdhuman_gts.py, the generated files are missing the v value in the line "i, x, y, w, h, v = map(float, (i, x, y, w, h, v))" during data reading.

Could I ask if you have encountered a similar situation before?

Thank you very much.

![image](https://github.com/user-attachments/assets/41533a90-55dc-4dd8-addf-3c632a04e975)
